test/clj_momo/lib/es/document_test.clj

```diff
@@ -288,11 +317,58 @@
     (is (= 3 (es-doc/count-docs conn "test_index" "test_mapping" {:ids {:values (range 3)}})))
     (es-index/delete! conn "test_index")))
 
-(deftest ^:integration query-test
+(deftest ^:integration scroll-test
   (let [sample-docs (mapv #(assoc {:_index "test_index"
                                    :_type "test_mapping"
                                    :foo :bar}
                                   :_id (str %))
+                          (range 100))
+        conn (es-conn/connect (th/get-es-config))
+        _ (es-index/delete! conn "test_index")
+        _ (es-index/create! conn "test_index" {})
+        _ (es-doc/bulk-create-doc conn sample-docs "true")
+        {batch-1 :data
+         {scroll-id-1 :scroll_id} :paging} (es-doc/query conn
+                                                         "test_index"
+                                                         "test_mapping"
+                                                         {:match_all {}}
+                                                         {:scroll "10s" :limit 40})
+       {batch-2 :data
+        {scroll-id-2 :scroll_id} :paging} (es-doc/scroll conn
+                                                         scroll-id-1
+                                                         {:scroll "10s"
+                                                          :full-hits? true})
+        {batch-3 :data
+         {scroll-id-3 :scroll_id} :paging} (es-doc/scroll conn
+                                                          scroll-id-2
+                                                          {:scroll "10s"})
+
+        {batch-4 :data
+         {scroll-id-4 :scroll_id} :paging} (es-doc/scroll conn
+                                                          scroll-id-3
+                                                          {})]
+    (is (= scroll-id-1
+           scroll-id-2
+           scroll-id-3
+           scroll-id-4))
+    (is (=  40 (count batch-1))
+        "the limit size must be applied on initial query")
+    (is (=  40 (count batch-2))
+        "the limit size must be preserved while scrolling")
+    (is (= (repeat 40 "test_index") (keep :_index batch-2))
+        "scroll should return document metadata")
+    (is (=  20 (count batch-3))
+        "scroll should retrieve partial last batch")
+
+    ;; clean
+    (es-index/delete! conn "test_index")))
```

Comment:
There is no test involving `scroll-4` except for the scroll-id. How can we know the scroll has finished?